src/plugins/persistence/couch/CouchObjectProvider.js

```diff
@@ -57,11 +57,19 @@ export default class CouchObjectProvider {
         return options;
     }
 
-    request(subPath, method, value) {
-        return fetch(this.url + '/' + subPath, {
-            method: method,
-            body: JSON.stringify(value)
-        }).then(response => response.json())
+    request(subPath, method, value, abortSignal) {
+        let fetchOptions = { method };
+
+        if (value) {
+            fetchOptions.body = JSON.stringify(value);
+        }
+
+        if (abortSignal) {
+            fetchOptions.signal = abortSignal;
+        }
```

Comment:
Can we simplify this as:
`fetchOptions = {method, value, signal}`
Since undefined values will just be ignored

Reply:
Yeah definitely